src/test/java/org/candlepin/resource/util/KeyValueStringParserTest.java

```diff
@@ -0,0 +1,616 @@
+        assertDoesNotThrow(() -> output1.add(elem));
+        assertEquals(1, output1.size());
+        assertTrue(output1.contains(elem));
+
+        List<Pair<String, String>> output2 = KeyValueStringParser.parseKeyValuePairs(List.of("key:value"));
+        assertNotNull(output2);
+
+        assertDoesNotThrow(() -> output2.add(elem));
+        assertEquals(2, output2.size());
+        assertTrue(output2.contains(elem));
+    }
+
+    @Test
+    public void testParseArrayPairs() {
+        String[] kvstrings = new String[] {
+            "key1:value1",
+            "key2:value2",
+            "key3:value3" };
+
+        Map<String, String> expected = new HashMap<>(Map.of(
+            "key1", "value1",
+            "key2", "value2",
+            "key3", "value3"));
+
+        List<Pair<String, String>> output = KeyValueStringParser.parseKeyValuePairs(kvstrings);
+        assertNotNull(output);
+        assertEquals(expected.size(), output.size());
+
+        for (Pair<String, String> pair : output) {
+            assertNotNull(pair);
+
+            assertTrue(expected.containsKey(pair.getKey()));
+            assertEquals(expected.remove(pair.getKey()), pair.getValue());
+        }
+
+        assertTrue(expected.isEmpty());
+    }
+
+    @Test
+    public void testParseArrayPairsWithMultipleDelimiters() {
+        String[] kvstrings = new String[] {
+            "key1:value1:extra",
+            "key2:value2:split:data",
+            "key3:value3:1:2:3" };
+
+        Map<String, String> expected = new HashMap<>(Map.of(
+            "key1", "value1:extra",
+            "key2", "value2:split:data",
+            "key3", "value3:1:2:3"));
+
+        List<Pair<String, String>> output = KeyValueStringParser.parseKeyValuePairs(kvstrings);
+        assertNotNull(output);
+        assertEquals(expected.size(), output.size());
+
+        for (Pair<String, String> pair : output) {
+            assertNotNull(pair);
+
+            assertTrue(expected.containsKey(pair.getKey()));
+            assertEquals(expected.remove(pair.getKey()), pair.getValue());
+        }
+
+        assertTrue(expected.isEmpty());
+    }
+
+    @Test
+    public void testParseArrayPairsPermitsNullCollection() {
+        List<Pair<String, String>> output = KeyValueStringParser.parseKeyValuePairs((String[]) null);
+        assertNotNull(output);
+        assertTrue(output.isEmpty());
+    }
+
+    @Test
+    public void testParseArrayPairsPermitsEmptyCollection() {
+        List<Pair<String, String>> output = KeyValueStringParser.parseKeyValuePairs();
+        assertNotNull(output);
+        assertTrue(output.isEmpty());
+    }
+
+    @Test
+    public void testParseArrayDisallowsNullValuesInCollection() {
+        String[] kvstrings = new String[] { "key1:value1", null, "key3:value3" };
+
+        assertThrows(BadRequestException.class, () -> KeyValueStringParser.parseKeyValuePairs(kvstrings));
+    }
+
+    @Test
+    public void testParseArrayDisallowsEmptyValuesInCollection() {
```

Comment:
You could add another test for a blank string as well. This can also apply to some of the other tests like where you are testing mapKeyValuePairs with an Iterable

Reply:
I don't understand the request. This test does handle empty strings already (parameter 2). If you're referencing a string containing whitespace (i.e. `"  "`), is there something the undelimited case isn't covering that you'd like to see?